### PR TITLE
Feature/LLM Moderation

### PR DESCRIPTION
**Pull request by HenryHengZJ**
added LLM to moderation to detect similar deny sentences

### CODE REVIEW

**Comment by HenryHengZJ** [comment]
image

File: packages/components/nodes/moderation/SimplePromptModeration/SimplePromptModerationRunner.ts [hunk 1]
```diff
@@ -1,23 +1,38 @@
...
     }
 
     async checkForViolations(input: string): Promise<string> {
-        this.denyList.split('\n').forEach((denyListItem) => {
-            if (denyListItem && denyListItem !== '' && input.toLowerCase().includes(denyListItem.toLowerCase())) {
-                throw Error(this.moderationErrorMessage)
+        if (this.model) {
+            const denyArray = this.denyList.split('\n')
+            for (const denyStr of denyArray) {
+                const res = await this.model.invoke(
```
Comment (vinodkiran) [inline packages/components/nodes/moderation/SimplePromptModeration/SimplePromptModerationRunner.ts:22]: denyString can be empty string, if there are trailing \n

File: packages/components/nodes/moderation/SimplePromptModeration/SimplePromptModerationRunner.ts [hunk 2]
```diff
@@ -1,23 +1,38 @@
...
+                    `Are these two sentences similar to each other? Only return Yes or No.\nFirst sentence: ${input}\nSecond sentence: ${denyStr}`
```
Comment (vinodkiran) [inline packages/components/nodes/moderation/SimplePromptModeration/SimplePromptModerationRunner.ts:23]: can we not send the entire denyList and make one LLM call with an adjusted prompt?

File: packages/components/nodes/moderation/SimplePromptModeration/SimplePromptModerationRunner.ts (hunk 1 above)
Comment (HenryHengZJ) [reply packages/components/nodes/moderation/SimplePromptModeration/SimplePromptModerationRunner.ts:23]: updated check

File: packages/components/nodes/moderation/SimplePromptModeration/SimplePromptModerationRunner.ts (hunk 2 above)
Comment (HenryHengZJ) [reply packages/components/nodes/moderation/SimplePromptModeration/SimplePromptModerationRunner.ts:24]: we can add an option to let user select if they want to make one LLM call per deny string, or all together at once

Comment (HenryHengZJ) [reply packages/components/nodes/moderation/SimplePromptModeration/SimplePromptModerationRunner.ts:24]: been trying to make a call like this: 
`const res = await this.model.invoke(
                    `Is this sentence: "${input}" similar to any of the sentences below? Only return Yes or No.\nSentences: ${this.denyList}`
                )`
However it always return false..

Comment (HenryHengZJ) [reply packages/components/nodes/moderation/SimplePromptModeration/SimplePromptModerationRunner.ts:24]: will let it run one LLM call per deny for now, as i cant get consistent performance when using the whole list

**Comment by HenryHengZJ** [comment]
@vinodkiran if you can take a final review again

**Pull request merged**